graylog2-server/src/main/java/org/graylog2/alerts/AbstractAlertCondition.java

```diff
@@ -155,6 +161,25 @@ public boolean shouldRepeatNotifications() {
         return repeatNotifications;
     }
 
+    /**
+     * Combines the given stream ID and query string into a single filter string.
+     *
+     * @param streamId the stream ID
+     * @param query    the query string (might be null or empty)
+     * @return the combined filter string
+     */
+    protected String buildQueryFilter(String streamId, String query) {
+        checkArgument(!isNullOrEmpty(streamId), "streamId parameter cannot be null or empty");
+
+        final StringBuilder builder = new StringBuilder().append("streams:").append(streamId);
+
+        if (!isNullOrEmpty(query) && !"*".equals(query)) {
```

Comment:
`CK_QUERY_DEFAULT_VALUE` and maybe trim?

Reply:
I don't want to use the configuration constant here because the filter builder would break if the default value changes. I updated the code to trim the streamId and query, though. Good catch, thanks! :+1: